### PR TITLE
add configuration code to policy controller

### PR DESCRIPTION
**Pull request by phoracek**
Run `nmstatectl set` directly from the NNCP controller. Tests will be adjusted
in follow-up PR since the scale is just too big to easily review in a single PR.

Signed-off-by: Petr Horacek <phoracek@redhat.com>

### CODE REVIEW

File: pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller.go [hunk 1]
```diff
@@ -73,13 +75,19 @@ func forThisNodePredicate(cl client.Client) predicate.Funcs {
...
 		DeleteFunc: func(deleteEvent event.DeleteEvent) bool {
-			return nodeSelectorMatchesThisNode(cl, deleteEvent.Object)
+			return false
 		},
 		UpdateFunc: func(updateEvent event.UpdateEvent) bool {
-			return nodeSelectorMatchesThisNode(cl, updateEvent.ObjectNew)
-		},
-		GenericFunc: func(genericEvent event.GenericEvent) bool {
-			return nodeSelectorMatchesThisNode(cl, genericEvent.Object)
+			if !nodeSelectorMatchesThisNode(cl, updateEvent.ObjectNew) {
```
Comment (qinqon) [inline pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller.go]: We are missing a lot of stuff from https://github.com/nmstate/kubernetes-nmstate/blob/master/pkg/controller/nodenetworkstateconfiguration/nodenetworkstateconfiguration_controller.go#L70-L94

File: pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller.go [hunk 2]
```diff
@@ -135,27 +143,91 @@ func (r *ReconcileNodeNetworkConfigurationPolicy) Reconcile(request reconcile.Re
...
-	nodeNetworkStateKey := types.NamespacedName{Name: nodeName}
-	err = r.client.Get(context.TODO(), nodeNetworkStateKey, nodeNetworkState)
+	nmstateOutput, err := nmstate.ApplyDesiredState(instance.Spec.DesiredState)
 	if err != nil {
-		if errors.IsNotFound(err) {
-			reqLogger.Info(fmt.Sprintf("the NodeNetworkState for %s is not there yet, let's requeue", nodeName))
-			// If there is no NodeNetworkState let's requeue could be that
-			// we are in the middle of the creation
-			return reconcile.Result{Requeue: true}, nil
+		errmsg := fmt.Errorf("error reconciling nodenetworkstate at desired state apply: %s, %v", nmstateOutput, err)
```
Comment (qinqon) [inline pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller.go]: s/nodenetworkstate/nodenetworkconfigurationpolicy/

File: pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller.go (hunk 1 above)
Comment (phoracek) [reply pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller.go]: The reason is that with the new design we reapply re-reconcile the configuration periodically, the check whether we need to Update or not should haven via the dry-run we discussed. We need to compare the desired configuration with the state of host, not with the previous desired configuration.

File: pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller.go (hunk 2 above)
Comment (phoracek) [reply pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller.go]: Fixed

File: pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller.go (hunk 1 above)
Comment (qinqon) [reply pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller.go]: Ahh that's right, ok.

**Pull request merged**